packages/rum/src/tracer.ts

```diff
@@ -0,0 +1,105 @@
+import { Configuration, XhrContext } from '@datadog/browser-core'
+
+export interface Tracer {
+  traceXhr: (context: Partial<XhrContext>, xhr: XMLHttpRequest) => TraceIdentifier | undefined
+}
+
+interface TracingHeaders {
+  [key: string]: string
+}
+
+export function startTracer(configuration: Configuration): Tracer {
+  return {
+    traceXhr: (context, xhr) =>
+      onTracingAllowed(context.url!, (tracingHeaders: TracingHeaders) => {
+        Object.keys(tracingHeaders).forEach((name) => {
+          xhr.setRequestHeader(name, tracingHeaders[name])
+        })
+      }),
+  }
+}
+
+function onTracingAllowed(url: string, inject: (tracingHeaders: TracingHeaders) => void): TraceIdentifier | undefined {
```

Comment:
The `on` prefix is usually used for event handlers. Maybe this function could be renamed to something more accurate... `injectHeadersIfTracingAllowed`.